resolver/type_syntax/type_syntax.cc

```diff
@@ -1184,10 +1184,30 @@ optional<TypeSyntax::ResultType> getResultTypeAndBindWithSelfTypeParamsImpl(core
                     result.type = core::Types::untypedUntracked();
                 }
             } else {
-                // a type member has occurred in a context that doesn't allow them
-                if (auto e = ctx.beginError(i.loc, core::errors::Resolver::InvalidTypeDeclaration)) {
-                    auto flavor = isTypeTemplate ? "type_template"sv : "type_member"sv;
-                    e.setHeader("`{}` `{}` is not allowed in this context", flavor, sym.show(ctx));
+                switch (args.typeMember) {
+                    case TypeSyntaxArgs::TypeMember::Allowed:
```

Comment:
Could we move the body of the `if` case into here, or get rid of the `switch` and then have an `if/else if/else`? It seems weird to have this hybrid of `if` and `switch`.

Reply:
Sure, I can do this. Originally I did not do this because I did not want to make the diff larger than it was.